src/main/java/com/google/cloud/discotoproto3converter/proto3/Message.java

```diff
@@ -41,39 +39,26 @@
     PRIMITIVES.put("", new Message("", false, true, null));
   }
 
-  private final String name;
-  private final List<Field> fields = new ArrayList<>();
+  private final SortedSet<Field> fields = new TreeSet<>();
   private final boolean ref;
-  // Not used for now, and hopefully will never be used.
-  private final List<Message> enums = new ArrayList<>();
-  private boolean isEnum;
+  private final SortedSet<Message> enums = new TreeSet<>();
+  private final boolean isEnum;
 
   public Message(String name, boolean ref, boolean isEnum, String description) {
-    super(description);
-    this.name = name;
+    super(name, description);
     this.ref = ref;
     this.isEnum = isEnum;
   }
 
-  public List<Field> getFields() {
+  public SortedSet<Field> getFields() {
     return fields;
   }
 
+  // NOTE: the fields list is expected to be sorted
```

Comment:
Which list? Maybe it would be clearer to say: "Precondition: The elements of this.fields are ordered".  (And is this not automatic given the use of TreeSet and SortedSet?)